courseworks/week2/Tinho_WHU/week2_1.md

```diff
@@ -0,0 +1,12 @@
+|                            | is concrete | is primitive | is abstract | is bits type | is mutable |
+| -------------------------- | ----------- | ------------ | ----------- | ------------ |:---------- |
+| ComplexF64                 | 1           | 0            | 0           | 1            | 1          |
+| Complex{AbstractFloat}     | 1           | 0            | 0           | 0            | 1          |
+| Complex{<:AbstractFloat}   | 0           | 0            | 0           | 0            | 0          |
+| AbstractFloat              | 0           | 0            | 1           | 0            | 0          |
+| Union{Float64, ComplexF64} | 0           | 0            | 0           | 0            | 0          |
+| Int32                      | 1           | 1            | 0           | 1            | 1          |
+| Matrix{Float32}            | 1           | 0            | 0           | 0            | 1          |
+| Base.RefValue              | 0           | 0            | 0           | 0            | 0          |
+
+1 for yes, 0 for no

```

Comment:
there are a few mistakes, you may check that using the functions in julia using
func = [Base.isconcretetype, Base.isprimitivetype, Base.isabstracttype, Base.isbitstype, Base.ismutabletype]